src/v/raft/consensus.cc

```diff
@@ -2478,6 +2478,27 @@ model::offset consensus::get_latest_configuration_offset() const {
     return _configuration_manager.get_latest_offset();
 }
 
+result<bool> consensus::is_under_replicated(model::node_id node_id) const {
+    if (!is_leader()) {
+        return errc::not_leader;
+    }
+    auto it = std::find_if(
+      _fstats.begin(), _fstats.end(), [node_id](const auto& s) {
+          return s.first.id() == node_id;
+      });
+    if (it == _fstats.end()) {
+        return errc::node_does_not_exists;
+    }
+
+    auto dirty_offset = _log.offsets().dirty_offset;
+    auto now = clock_type::now();
+    auto last_hbeat = it->second.last_hbeat_timestamp;
+    auto is_live = last_hbeat + _jit.base_duration() > now;
```

Comment:
The `is_live` check is valid, but also a bit weird in this context: we're checking that the leader has sent a heartbeat and got a prompt response recently, but we /know/ that our follower node is live because it's the one querying for his own under_replicated status.

If the follower finds that the leader doesn't perceive them as live, then maybe that's more something to emit a log warning for?  Like "node 123 queried under_replicated, but their last heartbeat response was 456ms ago".

Reply:
This is basically copy-pasted from `get_follower_metrics()`, just below, I should probably factor it out, or just call `get_follower_metrics().under_replicated`